lsp/nls/src/main.rs

```diff
@@ -33,6 +36,9 @@ struct Opt {
     /// The trace output file, disables tracing if not given
     #[arg(short, long)]
     trace: Option<PathBuf>,
+
+    #[arg(long)]
+    main_server: Option<String>,
```

Comment:
It's not user-facing so not very important, but I find it a bit surprising that (IIUC) starting `nls` with `--main-server` means that it just starts a background process to eval stuff. I would rather assume `--main-server` to be the main process.

Reply:
(Another solution might be to use [fork](https://docs.rs/fork/latest/fork/))

Reply:
Ah, but `--main-server` takes an argument so the background process gets run with `nls --main-server=foo`...

The fork crate looks nice, but I'm a little worried that it doesn't mention windows at all. I guess at some point I'll need to set up a windows vm to test on...

Reply:
Ok. Might be worth adding a description then.